Database issue ith Job Description of a submitted workflow is too large

If a workflow containing a large description field is submitted, the following exception is raised:

`[2017-11-13 10:30:56,360 6-thread-1 WARN    o.h.e.j.s.SqlExceptionHelper] SQL Error: 3401, SQLState: 22001
[2017-11-13 10:30:56,360 6-thread-1 ERROR   o.h.e.j.s.SqlExceptionHelper] data exception: string data, right truncation;  table: JOB_DATA column: DESCRIPTION
[2017-11-13 10:30:56,360 6-thread-1 ERROR              o.h.i.SessionImpl] HHH000346: Error during managed flush [could not execute statement]
[2017-11-13 10:30:56,361 6-thread-1 WARN       o.o.p.d.TransactionHelper] Database operation failed
org.hibernate.exception.DataException: could not execute statement
        at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:52)
        at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:42)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:111)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:97)
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:207)
        at org.hibernate.engine.jdbc.batch.internal.NonBatchingBatch.addToBatch(NonBatchingBatch.java:45)
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:2921)
        at org.hibernate.persister.entity.AbstractEntityPersister.insert(AbstractEntityPersister.java:3421)
        at org.hibernate.action.internal.EntityInsertAction.execute(EntityInsertAction.java:89)
        at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:560)
        at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:434)
        at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:337)
        at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:39)
        at org.hibernate.internal.SessionImpl.flush(SessionImpl.java:1295)
        at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:468)
        at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:3135)
        at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2352)
        at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:485)
        at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:147)
        at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$100(JdbcResourceLocalTransactionCoordinatorImpl.java:38)
        at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:231)
        at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:65)
        at org.ow2.proactive.db.TransactionHelper.tryExecuteTransaction(TransactionHelper.java:124)
        at org.ow2.proactive.db.TransactionHelper.tryExecuteTransactionLoop(TransactionHelper.java:96)
        at org.ow2.proactive.db.TransactionHelper.executeReadWriteTransaction(TransactionHelper.java:78)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.executeReadWriteTransaction(SchedulerDBManager.java:1729)
        at org.ow2.proactive.scheduler.core.db.SchedulerDBManager.newJobSubmitted(SchedulerDBManager.java:1579)
        at org.ow2.proactive.scheduler.core.LiveJobs.jobSubmitted(LiveJobs.java:283)
        at org.ow2.proactive.scheduler.core.SubmitHandler.run(SubmitHandler.java:51)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.sql.SQLDataException: data exception: string data, right truncation;  table: JOB_DATA column: DESCRIPTION
        at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
        at org.hsqldb.jdbc.JDBCPreparedStatement.fetchResult(Unknown Source)
        at org.hsqldb.jdbc.JDBCPreparedStatement.executeUpdate(Unknown Source)
        at com.zaxxer.hikari.proxy.PreparedStatementProxy.executeUpdate(PreparedStatementProxy.java:61)
        at com.zaxxer.hikari.proxy.PreparedStatementJavassistProxy.executeUpdate(PreparedStatementJavassistProxy.java)
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:204)
        ... 29 more`

FYI, the following description raised this issue:
`This workflow shows how to use ProActive Cloud Automation to automate the deployment and the life-cycle management of third-party services, required by some tasks.
Here, VisdomTask wants to send some data to a Visdom service.
To do so, it depends on bind_or_start_service (you can get it from the Examples/5. Cloud Automation Task Templates), where we have specified the service we want to start, and its instance name, which is inherited by a job variable.
Thus, VisdomTask can use a “endpoint” variable where the Visdom service is running. Everything is handled by the previous bind_or_start_service task.
Also, If you open the result of bind_or_start_service in a browser, you are automatically redirected your Visdom instance. You can also retrieve this URL from the ProActive Cloud Automation portal.
Once VisdomTask is finished, we go through a Validation step so the Visdom keeps running and the user can still visualize his results.
Go to the notification portal and validate to terminate the service with the task “terminate_service” and then the, job properly.
Please note that you can reuse an existing Visdom instance if you reuse the same instance name when you submit the job.`

But the following does not:
`This workflow shows how to use ProActive Cloud Automation to automate the deployment and the life-cycle management of third-party services, required by some tasks.
Here, VisdomTask wants to send some data to a Visdom service.
To do so, it depends on bind_or_start_service (you can get it from the Examples/5. Cloud Automation Task Templates), where we have specified the service we want to start, and its instance name, which is inherited by a job variable.
Thus, VisdomTask can use a “endpoint” variable where the Visdom service is running. Everything is handled by the previous bind_or_start_service task.
Also, If you open the result of bind_or_start_service in a browser, you are automatically redirected your Visdom instance. You can also retrieve this URL from the ProActive Cloud Automation portal.
Once VisdomTask is finished, we go through a Validation step so the Visdom keeps running and the user can still visualize his results.`

--- a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobContent.java
+++ b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobContent.java
@@ -46,6 +46,7 @@ import org.apache.commons.lang3.SerializationUtils;
 import org.apache.log4j.Logger;
 import org.hibernate.annotations.GenericGenerator;
 import org.hibernate.annotations.Parameter;
+import org.hibernate.annotations.Type;
 import org.ow2.proactive.scheduler.common.job.Job;
 import org.ow2.proactive.scheduler.common.job.TaskFlowJob;
 import org.ow2.proactive.scheduler.util.ByteCompressionUtils;
@@ -67,6 +68,7 @@ public class JobContent implements Serializable {
     private static final Logger LOGGER = Logger.getLogger(JobContent.class);
 
     @Lob
+    @Type(type = "org.hibernate.type.BinaryType")
     @Column(name = "CONTENT", length = Integer.MAX_VALUE)
     private byte[] jobContentAsByteArray;
 

--- a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java
+++ b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/JobData.java
@@ -26,7 +26,6 @@
 package org.ow2.proactive.scheduler.core.db;
 
 import java.io.Serializable;
-import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -407,7 +406,8 @@ public class JobData implements Serializable {
         this.userSpace = userSpace;
     }
 
-    @Column(name = "DESCRIPTION", length = 1000, updatable = false)
+    @Lob
+    @Column(name = "DESCRIPTION", length = Integer.MAX_VALUE, updatable = false)
     public String getDescription() {
         return description;
     }

--- a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java
+++ b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskData.java
@@ -861,7 +861,8 @@ public class TaskData {
         }
     }
 
-    @Column(name = "DESCRIPTION", updatable = false)
+    @Lob
+    @Column(name = "DESCRIPTION", length = Integer.MAX_VALUE, updatable = false)
     public String getDescription() {
         return description;
     }

--- a/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskResultData.java
+++ b/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/TaskResultData.java
@@ -175,8 +175,9 @@ public class TaskResultData {
         this.resultTime = resultTime;
     }
 
-    @Column(name = "RESULT_VALUE", length = Integer.MAX_VALUE)
     @Lob
+    @Type(type = "org.hibernate.type.BinaryType")
+    @Column(name = "RESULT_VALUE", length = Integer.MAX_VALUE)
     public byte[] getSerializedValue() {
         return serializedValue;
     }
@@ -185,8 +186,9 @@ public class TaskResultData {
         this.serializedValue = serializedValue;
     }
 
-    @Column(name = "RESULT_EXCEPTION", length = Integer.MAX_VALUE)
     @Lob
+    @Type(type = "org.hibernate.type.BinaryType")
+    @Column(name = "RESULT_EXCEPTION", length = Integer.MAX_VALUE)
     public byte[] getSerializedException() {
         return serializedException;
     }

